src/Microsoft.Restier.Providers.EntityFramework/Model/ModelProducer.cs

```diff
@@ -62,7 +62,19 @@ public Task<IEdmModel> GetModelAsync(ModelContext context, CancellationToken can
             var dbContext = context.GetApiService<DbContext>();
 
             var efModel = (dbContext as IObjectContextAdapter).ObjectContext.MetadataWorkspace;
-            var efEntityContainer = efModel.GetItems<EntityContainer>(DataSpace.CSpace).Single();
+            var efEntityContainers = efModel.GetItems<EntityContainer>(DataSpace.CSpace);
+            var efEntityContainer = efEntityContainers.FirstOrDefault(c => c.Name == dbContext.GetType().Name);
+            if (efEntityContainer == null)
+            {
+                if (efEntityContainers.Count > 1)
+                {
```

Comment:
Done. Hopefully those help. I will see about moving the exception messages to the resources over the weekend.